### DISCUSSION TITLE
Seeking Help: Keeping the Display Always On for My Heart Rate Zone App on T-Rex 2

### DISCUSSION THREAD

**Discussion by mohansella** [zepp-health/zeppos-forum#396] [Q&A] [answered]
Hi everyone,

I’m working on a [personal project](https://github.com/mohansella/zepp-hr-zone/tree/fbcb4923f2440aefa2b91f163542d25fd8ede844) to create a guided fat-burning heart rate zone app for the T-Rex 2 watch, which supports v2.1 APIs. The app vibrates when the heart rate exceeds or falls below the fat-burn zone. I developed this to avoid checking my wrist every 5-10 seconds while swimming.

I thought I had completed the project, but I’ve encountered an issue: I can’t find a way to keep the display always on. Currently, the display turns off after 30 seconds regardless of what I do, and the app freezes until the screen is woken again. This is really frustrating.

I even tried enabling the screen-on mode during workouts, but it doesn’t work for my app (though the system app for swimming seems to keep the display on).

Here are my questions:

How can I program my app to ensure the display stays on while it’s open?
If this isn’t supported, is there any workaround? Any solution is fine, as my primary goal is to make the watch vibrate when my heart rate leaves the fat-burn zone.
Any guidance or suggestions would be greatly appreciated. Thanks in advance!

**Comment by silver-zepp** [comment]
```js
// don't turn off the screen for 600 seconds
import { setPageBrightTime } from '@zos/display'
const result = setPageBrightTime({
  brightTime: 600 * 1000,
})

// don't turn off the screen on wrist down for 600 seconds
import { pauseDropWristScreenOff } from '@zos/display'
pauseDropWristScreenOff({
  duration: 600 * 1000,
})
```

**Comment by mohansella** [comment, marked as answer]
looks like below is the solution that is working for me,
```   javascript
    pauseDropWristScreenOff({ duration: 0 })
    pausePalmScreenOff({ duration: 0 })
    setWakeUpRelaunch({ relaunch: true })
```
